docs_src/design/adr/security/0008-Secret-Creation-and-Distribution.md

```diff
@@ -0,0 +1,259 @@
+   inject secrets into the process environment,
+   and then exec's a replacment process
+   that inherits the secret-enriched enviornment block.
+
+   There are a few potential risks with this approach:
+
+   * Environment blocks are passed to child processes by default.
+   * Environment-variable-sniffing malware (introduced by compromised 3rd party libaries) is a proven attack method.
+
+3. Dynamic injection of secret into container-scoped `tmpfs` volume
+
+   An example of this approach is [consul-template](https://github.com/hashicorp/consul-template).
+   This approach is useful when a secret is required to be in a configuration file
+   and cannot be passed via an environment variable
+   or directly consumed from a secret store.
+
+4. Distribution via a secrets volume
+
+   This option is the most widely supported secret distribution mechanism by container orchestrators.
+
+   EdgeX supports runtime environments such as standard Docker and snaps
+   that have no built-in secret management features.
+   EdgeX must therefore implement its own compatible secrets mechanism.
+
+   * Docker Swarm:
+   Swarm swarm mode is not officially supported by the EdgeX project.
+   Docker Swarm secrets are shared via the `/run/secrets` volume,
+   which is a Linux `tmpfs` volume created on the host and shared with the container.
+   For an example of Docker Swarm secrets, see the
+   [docker-compose secrets stanza](https://docs.docker.com/compose/compose-file/#secrets).
+   Secrets distributed in this manner become part of the RaftDB,
+   and thus it becomes necessary to enable swarm autolock mode,
+   which prevents the Raft database encryption key
+   from being stored plaintext on disk.
+
+   * Kubernetes:
+   Kubernetes is not officially supported by the EdgeX project.
+   Kubernetes also supports the secrets volume approach,
+   though the secrets volume can be mounted anywhere in the container namespace.
+   For an example of Kubernetes secrets volumes, see the
+   [Kubernetes secrets documentation](https://kubernetes.io/docs/concepts/configuration/secret/).
+   Secrets distributed in this manner become part of the `etcd` database,
+   and thus it becomes necessary to specify a
+   [KMS provider for data encryption](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/)
+   to prevent `etcd` from storing plaintext versions of secrets.
+
+   * Docker:
+   Use of a RAM-based temporary file system is desirable to protect secrets,
+   but only Linux platforms support `tmpfs` volumes for secret sharing:
+   generally, `/run` on a Linux system is a `tmpfs` volume.
+   For reasons of portability, `/tmp/edgex/secrets` has been chosen as a
+   cross-platform alternative to `/run`:
+   on the MacOS Docker implementation,
+   `/tmp` is a white-listed shared host directory,
+   while `/run` is not.
+   For production hardening, `/tmp` should be replaced by `/run`
+   where it appears in `docker-compose` files.
```

Comment:
Huuum, we should make this a variable that is set in the new .env file. 

TMPFS_VOLUME=/tmp

Reply:
Good suggestion.  I will try it once the revised compose files are committed.